src/http/showcase.rs

```diff
@@ -0,0 +1,102 @@
+use std::sync::Arc;
+
+use anyhow::{anyhow, Result};
+use async_graphql::ServerError;
+use hyper::{Body, Request, Response};
+use serde::de::DeserializeOwned;
+use url::Url;
+
+use super::AppContext;
+use crate::async_graphql_hyper::{GraphQLRequestLike, GraphQLResponse};
+use crate::blueprint::Blueprint;
+use crate::cli::{init_chrono_cache, init_http};
+use crate::config::reader::ConfigReader;
+use crate::config::Upstream;
+use crate::{EnvIO, FileIO, HttpIO};
+
+struct DummyFileIO;
+
+impl FileIO for DummyFileIO {
```

Comment:
for a custom file IO. We can load configuration over HTTP anyways via `config::reader`, we should leverage that.

Reply:
`ConfigReader` requires a `File` handle and will attempt to read from the file system if it cannot parse the URL. This is needed in `cloudflare/src/handle.rs`, but it could be dangerous in a showcase environment. I implemented this to allow disabling the `File` operations without uprooting `ConfigReader`.

I can make the change in `ConfigReader` to take an `Option<File>` instead of this if you'd like, but I'd need to change a lot of unrelated codepaths.

Reply:
> I can make the change in `ConfigReader` to take an `Option<File>` instead of this if you'd like, but I'd need to change a lot of unrelated codepaths.

Actually, this is not even possible. We need to provide `showcase_get_app_ctx` with a type implementing `FileIO` even when we're not using it, because using `Option<dyn FileIO>` causes a type-checking cycle and a build fail. We could give it `NativeFileIO` as a type, but that breaks WASM build, because it will depend on `cli`. Thus `DummyFileIO` is needed.